tencentcloud/resource_tc_instance.go

```diff
@@ -273,6 +288,18 @@ func resourceTencentCloudInstanceCreate(d *schema.ResourceData, m interface{}) e
 			params["InstanceName"] = insName
 		}
 	}
+	if hostName, ok := d.GetOk("hostname"); ok {
+		hName := hostName.(string)
+		if len(hName) > 0 {
+			params["HostName"] = hName
+		}
+	}
+	if projectId, ok := d.GetOk("project_id"); ok {
+		pId := projectId.(int)
+		if pId > 0 {
```

Comment:
project id can be 0, and it is the default value, leave this check to API side

Reply:
Fixed, let server validate.